experimental/GModule/Cohomology.jl

```diff
@@ -1381,13 +1484,62 @@ iszero(a) || (@show g, h, k, a ; return false)
   return true
 end
 
+"""
+Computes H^3 via dimension-shifting:
+There is a short exact sequence
+  1 -> A -> Hom(Z[G], A) -> B -> 1 
+thus
+  H^3(G, A) = H^2(C, B)
+as Hom(Z[G], A) is induced hence has trivial cohomology.
+Currently only the group is returned
+"""
+function H_three(C::GModule{<:Oscar.GAPGroup, <:Any})
+  G = C.G
+  if isa(C.M, GrpAbFinGen)
+    zg, ac, em = Oscar.GModuleFromGap.natural_gmodule(GrpAbFinGen, G, ZZ)
+  elseif isa(C.M, AbstractAlgebra.FPModule{<:FieldElem})
+    zg, ac, em = Oscar.GModuleFromGap.natural_gmodule(G, base_ring(C))
+  else
+    error("unsupported module")
+  end
+  @assert is_consistent(zg)
+  H, mH = Oscar.GModuleFromGap.ghom(zg, C)
+  @assert is_consistent(H)
+  #from https://math.mit.edu/classes/18.786/2018/LectureNotes29.pdf
+  #around 29.8
+  #Drew's notes.
+  #the augmentation map on the (canonical) generators is 1
+  inj = hom(C.M, H.M, [preimage(mH, hom(zg.M, C.M, [c for g = gens(zg.M)])) for c = gens(C.M)])
+  @assert is_G_hom(C, H, inj)
+  q, mq = quo(H, image(inj)[2])
+#  return q, mq, inj, H
+  #possibly, to get 3-chains: 
+  # 2 chain in q
+  # preimage mq 2 chain in H
+  # differential 3 chain in H
+  # preimage inj 3 chain in C
+  return H_two(q)[1]
+end
+
+function is_right_G_module(C::GModule)
+  #tests if the action is right-linear 
+  G = C.G
+  return all([action(C, g)*action(C, h) == action(C, g*h) for g = gens(G) for h = gens(G)])
```

Comment:
Don't you mean
```suggestion
  return all(action(action(C, g), h) == action(C, g*h) for g in gens(G), h in gens(G))
```
(This also avoids the allocation of a vector plus it can abort on the first violation)